File: src/nuzzgraph/dbclient/NodeHelper.java
package nuzzgraph.dbclient;

import com.tinkerpop.blueprints.pgm.Vertex;

/**
 * User: Mark Nuzzolilo
 * Date: 3/21/12
 * Time: 1:18 AM
 */
public class NodeHelper
{
    static int vertexClusterId;

    public static int getNodeId(Vertex v)
    {
        String vId = v.getId().toString();
        return Integer.parseInt(vId.substring(vId.indexOf(":") + 1));
    }
    
    public static String getVertexId(int nodeId)
    {
        return "#" + vertexClusterId + ":" + nodeId;
    }
}

File: src/nuzzgraph/dbclient/nuzzgraph.dbclient.main/NodeHelper.scala
package nuzzgraph.dbclient.main

import com.tinkerpop.blueprints.pgm.Vertex

/**
 * User: Mark Nuzzolilo
 * Date: 3/21/12
 * Time: 1:18 AM
 */
object NodeHelper {
  def getNodeId(v: Vertex): Int = {
    var vId: String = v.getId.toString
    return Integer.parseInt(vId.substring(vId.indexOf(":") + 1))
  }

  def getVertexId(nodeId: Int): String = {
    return "#" + vertexClusterId + ":" + nodeId
  }

  private[dbclient] var vertexClusterId: Int = 0
}
